cmd/otel-agent/subcommands/status/command.go

```diff
@@ -0,0 +1,240 @@
+			cliParams.args = args
+
+			// Prevent autoconfig to run when running status as it logs before logger
+			// is setup.  Cannot rely on config.Override as env detection is run before
+			// overrides are set.  TODO: This should eventually be handled with a
+			// BundleParams field for AD.
+			os.Setenv("DD_AUTOCONFIG_FROM_ENVIRONMENT", "false")
+
+			return fxutil.OneShot(statusCmd,
+				fx.Supply(cliParams),
+				fx.Supply(core.BundleParams{
+					ConfigParams: config.NewAgentParams(globalParams.CoreConfPath, config.WithExtraConfFiles(globalParams.ConfPaths)),
+					LogParams:    log.ForOneShot(command.LoggerName, cliParams.logLevelDefaultOff.Value(), true)}),
+				core.Bundle(),
+			)
+		},
+	}
+	cliParams.logLevelDefaultOff.Register(cmd)
+	cmd.PersistentFlags().BoolVarP(&cliParams.jsonStatus, "json", "j", false, "print out raw json")
+	cmd.PersistentFlags().BoolVarP(&cliParams.prettyPrintJSON, "pretty-json", "p", false, "pretty print JSON")
+	cmd.PersistentFlags().StringVarP(&cliParams.statusFilePath, "file", "o", "", "Output the status command to a file")
+	cmd.PersistentFlags().BoolVarP(&cliParams.verbose, "verbose", "v", false, "print out verbose status")
+	cmd.PersistentFlags().BoolVarP(&cliParams.list, "list", "l", false, "list all available status sections")
+
+	return cmd
+}
+
+func scrubMessage(message string) string {
+	msgScrubbed, err := scrubber.ScrubBytes([]byte(message))
+	if err == nil {
+		return string(msgScrubbed)
+	}
+	return "[REDACTED] - failure to clean the message"
+}
+
+func redactError(unscrubbedError error) error {
+	if unscrubbedError == nil {
+		return unscrubbedError
+	}
+
+	errMsg := unscrubbedError.Error()
+	scrubbedMsg, scrubOperationErr := scrubber.ScrubBytes([]byte(errMsg))
+	var scrubbedError error
+	if scrubOperationErr != nil {
+		scrubbedError = errors.New("[REDACTED] failed to clean error")
+	} else {
+		scrubbedError = errors.New(string(scrubbedMsg))
+	}
+
+	return scrubbedError
+}
+
+func statusCmd(logger log.Component, config config.Component, _ sysprobeconfig.Component, cliParams *cliParams) error {
+	if cliParams.list {
+		return redactError(requestSections(config))
+	}
+
+	if len(cliParams.args) < 1 {
+		return redactError(requestStatus(config, cliParams))
+	}
+
+	return componentStatusCmd(logger, config, cliParams)
+}
+
+func setIpcURL(cliParams *cliParams) url.Values {
+	v := url.Values{}
+	if cliParams.verbose {
+		v.Set("verbose", "true")
+	}
+
+	if cliParams.prettyPrintJSON || cliParams.jsonStatus {
+		v.Set("format", "json")
+	} else {
+		v.Set("format", "text")
+	}
+
+	return v
+}
+
+func renderResponse(res []byte, cliParams *cliParams) error {
+	var s string
+
+	// The rendering is done in the client so that the agent has less work to do
+	if cliParams.prettyPrintJSON {
+		var prettyJSON bytes.Buffer
+		json.Indent(&prettyJSON, res, "", "  ") //nolint:errcheck
+		s = prettyJSON.String()
+	} else if cliParams.jsonStatus {
+		s = string(res)
+	} else {
+		s = scrubMessage(string(res))
```

Comment:
Status should be scrubbed on the agent side already. @GustavoCaso do you confirm ?